fw/cache.c

```diff
@@ -1752,6 +1752,48 @@ __cache_add_node(TDB *db, TfwHttpResp *resp, unsigned long key)
 	}
 }
 
+/*
+ * Processing of reponse to PURGE with X-Tempesta-Cache header:
+ * 	Remove body content, set Content-Length to zero
+ */
+static void
+tfw_process_purge_x_tempesta_cache_response(TfwHttpResp *resp)
+{
+	TfwStr *dup, *end;
+	TfwStr replacement = {
+		.chunks = (TfwStr []) {
+			TFW_STR_STRING("Content-Length"),
+			TFW_STR_STRING(": "),
+			TFW_STR_STRING("0")
+		},
+		.nchunks = 3,
+	};
+	TfwStr *c = replacement.chunks;
+	int r;
+
+	if(!TFW_STR_EMPTY(&resp->body)) {
+		/* Delete the underlying body data. */
+		TFW_STR_FOR_EACH_DUP(dup, &resp->body, end) {
```

Comment:
We use DUP for duplicate stings, i.e. duplicate headers only. There is no duplicated body, so we can use simpler looping

Reply:
ok. done.